### DISCUSSION TITLE
mimxrt1050-evk: Hyperflash or QSPI flash?

### DISCUSSION THREAD

**Discussion by robert-hh** [adafruit/tinyuf2#415] [Q&A]
The UF2 implementation in mimxrt1050-evk supports QSPI flash. The MIMXRT1050-EVK board I have was delivered configured for Hyperflash, and according to the data sheet that is the default. I do not know how many people rewire the board for QSPI, but for me it's not feasible.
Is there a Hyperflash version of the UF2 bootloader available or planned?

**Comment by hathach** [comment]
is your 1050-evk or evkb version, iirc both of them has DIP switch to change the boot device. QSPI is implemented by default since it is the most likely flash device people would design board around.

![image](https://github.com/user-attachments/assets/bf56d5da-427a-4d6f-b9ab-5562ead67d29)

For adding hyper flash support, I think we will only need to change the bootrom flash driver, device control config and/or board's pins config. Most of the tinyuf2 code could be untouched. Let me know if you could make the DIP changes, otherwise I will also try to pull out my 1050-evkB to test with hyperflash later as well.

**Reply by tannewt** [reply]
The switching from hyperflash to qspi needs modification on the board. I did it on mine because I didn't want to learn hyperflash stuff. I think the 1050 EVK was the first in the series so they've changed their approach since then. I'd assume other boards with the 1050 would also be qspi. I don't think hyperflash is worth it.

**Reply by robert-hh** [reply]
The board i have is a evkb model. And yes, one need to rewire the board to switch from hyperflash to QSPI. It's a handful of 0-Ohm SMD resistors that have to be moved, size like 0402. Too small for me.
I have another 1052 board from  Seeed. The have QSPI as default and the connections for a hyperflash chip. But it is not populated.

**Reply by hathach** [reply]
thanks Scott for clarification. Sorry, my memory fades, I will try to see if I could get hyperflash running quickly when I get back from current work.
